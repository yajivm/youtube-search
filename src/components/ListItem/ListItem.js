import React from 'react';
import PropTypes from 'prop-types';
import { getPublishedDate } from '../../helpers/utils';

import './ListItem.style.css';

const ListItem = ({ listData }) => {

    return (
        <ul className="details-list" data-testid='details-list'>
            <li data-testid="channel-title">{listData?.channelTitle}</li>
            <li data-testid="published-date">{getPublishedDate(listData?.publishedAt)}</li>
        </ul>
    );
};

ListItem.propTypes = {
    channelTitle: PropTypes.string,
    publishedAt: PropTypes.string,
};

ListItem.defaultProps = {
    channelTitle: '',
    publishedAt: '',
};

export default ListItem;
