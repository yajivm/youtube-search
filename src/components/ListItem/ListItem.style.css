.details-list {
    list-style: none;
    font-size: 12px;
    margin: 0;
    padding: 0;
}

.details-list > li {
  display: inline;
  font-weight: 700;
  color: #939393;
}

.details-list > li:first-child::after {
    content: "â€¢";
    margin: 0 4px;
}
