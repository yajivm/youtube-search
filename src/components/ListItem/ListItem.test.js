import React from 'react';
import { render } from '@testing-library/react';

import ListItem from './ListItem';

const mockGetPublishedData = jest.fn();
jest.mock('../../helpers/utils', () => {
    return {
        getPublishedDate: (...args) => mockGetPublishedData(...args),
    }
});

const mockListData = {
    channelTitle: 'aChannelTestTitle',
    publishedAt: '2021-11-12T22:18:07Z',
};

test('should render list items', () => {
    const { getByTestId } = render(<ListItem listData={mockListData} />);
    
    expect(getByTestId('channel-title')).toBeDefined();
    expect(getByTestId('channel-title').textContent).toStrictEqual('aChannelTestTitle');
    expect(mockGetPublishedData).toHaveBeenCalledTimes(1);
    expect(mockGetPublishedData).toHaveBeenCalledWith('2021-11-12T22:18:07Z');
    expect(getByTestId('published-date')).toBeDefined();
    expect(getByTestId('details-list')).toMatchSnapshot();
});