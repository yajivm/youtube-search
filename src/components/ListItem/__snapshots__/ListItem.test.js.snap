// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list items 1`] = `
<ul
  class="details-list"
  data-testid="details-list"
>
  <li
    data-testid="channel-title"
  >
    aChannelTestTitle
  </li>
  <li
    data-testid="published-date"
  />
</ul>
`;
